db.go

```diff
@@ -339,6 +342,11 @@ func (db *DB) Merge() error {
 		return errors.New("the number of files waiting to be merged is at least 2")
 	}
 
+	//check the key of list is expired or not
+	//if expired, it will clear the items of index
+	//so that nutsdb can clear entry of expiring list in the function getPendingMergeEntries
+	db.checkListExpired()
```

Comment:
It is not appropriate to put it here, just filter out the keys of the expired list when rewriting

Reply:
If the list expires and needs to be checked, do other data structures also need to be checked?

Reply:
now ，we only have List structures need use this way to check，because other structures can check it by isFilterEntry().They use their timestamp and ttl to check it.Also, if we will add ttl for set, we should do it in this way.

Reply:
I think this logic can not put in rewrite.Because it is to clear the items in index,and then when getPendingMergeEntries(), it will delete the log that does not exist in the index.So we must put it behind getPendingMergeEntries(), then we can abort abandon the expire entry

Reply:
![image](https://user-images.githubusercontent.com/74813175/212447078-391bac42-a441-4fe8-9fd1-9c61efbab0b6.png)

Reply:
I think we can put it there

Reply:
ok i see